x/finality/keeper/msg_server.go

```diff
@@ -303,3 +309,39 @@ func (k Keeper) slashFinalityProvider(ctx context.Context, fpBtcPk *bbn.BIP340Pu
 		panic(fmt.Errorf("failed to emit EventSlashedFinalityProvider event: %w", err))
 	}
 }
+
+// validateActivationHeightAddFinalitySig returns error if the msg add finality
+// block height is lower than the activation height
+func (ms msgServer) validateActivationHeightAddFinalitySig(ctx sdk.Context, msg *types.MsgAddFinalitySig) error {
```

Comment:
```suggestion
func (ms msgServer) validateActivationHeight(ctx sdk.Context, height uint32) error {
```
think maybe we could make it general by just passing in the height. logs can be wrapped by the called for more context